Docs update for deploying with Kustomize (#47772)
sourcegraph/sourcegraph: doc/admin/deploy/kubernetes/kustomize/eks.md

# Installation Guide - Amazon Elastic Kubernetes Service (EKS)

This section is aimed at providing high-level guidance on deploying Sourcegraph using a Kustomize overlay on Amazon Elastic Kubernetes Service (EKS). 

## Overview

> The installation steps below will walk you through the steps using our quick-start overlay for Elastic Kubernetes Service (EKS) to deploy Sourcegraph.

Reviewer: ```suggestion
The installation instructions below will guide you through deploying Sourcegraph on Elastic Kubernetes Service (EKS) with our quick-start overlay.
```

- Minimum Kubernetes version: [v1.19](https://kubernetes.io/blog/2020/08/26/kubernetes-release-1.19-accentuate-the-paw-sitive/) with [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) v1.19 or later
- [Kustomize](https://kustomize.io/) (built into `kubectl` in version >= 1.14)

## Quick Start

You must complete **all** the prerequisites listed above before installing Sourcegraph with following steps.

### Step 1: Deploy Sourcegraph

Deploy Sourcegraph main app without the monitoring stacks to your cluster:

```bash
$ kubectl apply --prune -l deploy=sourcegraph -k https://github.com/sourcegraph/deploy-sourcegraph-k8s/examples/aws
```

Monitor the deployment status to make sure everything is up and running:

```bash
kubectl get pods -o wide --watch
```

### Step 2: Access Sourcegraph in Browser

To check the status of the load balancer and obtain its IP:

```bash
$ kubectl describe ingress sourcegraph-frontend
```

From you output, look for the IP address of the load balancer, which is listed under `Address`.

```bash
# Sample output:
Name:             sourcegraph-frontend
Namespace:        default
Address:          12.345.678.0
```

> Once the load balancer is ready, you can access your new Sourcegraph instance at the returned IP address in your browser via HTTP. Accessing the IP address with HTTPS returns errors because TLS must be enabled first.

Reviewer: ```suggestion
Once the load balancer is ready, you can access your new Sourcegraph instance at the returned IP address in your browser via HTTP. Accessing the IP address with HTTPS will return errors because TLS must be enabled first.
```

```bash
kubectl port-forward svc/sourcegraph-frontend 3080:30080
```

You should now be able to access your new Sourcegraph instance at http://localhost:3080  🎉

### Further configuration

The steps above have guided you to deploy Sourcegraph using the [quick-start/aws/eks](https://github.com/sourcegraph/deploy-sourcegraph-k8s/tree/master/examples/aws) overlay preconfigured by us.

If you would like to make other configurations to your existing instance, you can create a new overlay using its kustomization.yaml file shown below and build on top of it. For example, you can upgrade your instance from size XS to L, or add the monitoring stacks.

```yaml
# overlays/$INSTANCE_NAME/kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: default
resources:
# Sourcegraph Main Stack
- ../../base/sourcegraph
components:
# Sourcegraph Monitoring Stack
- ../../components/monitoring
# Use resources for a size-XS instance
- ../../components/sizes/xs
# Apply configurations for AWS EKS storage class and ALB
- ../../components/clusters/aws/eks-ebs
```

#### Enable TLS

> Once you have created a new overlay using the kustomization file from our quick-start overlay for AWS EKS, we strongly recommend you to: 

Reviewer: ```suggestion
Once you have created a new overlay using the kustomization file from our quick-start overlay for AWS EKS, we strongly recommend that you:
```
